aten/src/ATen/core/op_registration/kernel_function_test.cpp

```diff
@@ -30,21 +31,23 @@ C10_DEFINE_TENSOR_TYPE(TensorType1);
 C10_DECLARE_TENSOR_TYPE(TensorType2);
 C10_DEFINE_TENSOR_TYPE(TensorType2);
 
-void errorKernel(const Tensor&) {
+int64_t errorKernel(const Tensor& tensor, int64_t input) {
   EXPECT_TRUE(false); // this kernel should never be called
+  return 0;
 }
 
 FunctionSchema errorOpSchema(
     "_test::error",
     "",
-    (std::vector<Argument>{Argument("dummy")}),
-    (std::vector<Argument>{}));
+    (std::vector<Argument>{Argument("dummy"),
+                           Argument("input", IntType::get())}),
+    (std::vector<Argument>{Argument("output", IntType::get())}));
 
-int incrementKernel(const Tensor& tensor, int input) {
+int64_t incrementKernel(const Tensor& tensor, int64_t input) {
```

Comment:
Is this op_regsitration code shared with the same code in JIT operators? If not, why not? If so, is this breaking backward compatibility of that code by being more restrictive about the allowed integer input types?

Reply:
errorKernel is used in these test cases together with a schema that says the kernel is supposed to return IntType. This was actually a bug but not noticed since nothing checked the schema against the kernel function. Now with this diff, the schema is checked against the kernel function and this threw an error.

Reply:
The code inferring a schema from a C function is shared. The restriction towards `int64_t` comes from the existing JIT implementation. We can consider making it less restrictive in future.

Reply:
So will this change affect backward compatibility or not?

Reply:
No, it shouldn't. It's the very same restriction in the existing JIT custom op implementation and here because both share the same code.